DashAI/back/api/api_v1/endpoints/tasks.py

```diff
@@ -0,0 +1,55 @@
+import logging
+from collections import defaultdict
+
+from fastapi import APIRouter, status
+from fastapi.exceptions import HTTPException
+
+from DashAI.back.core.config import task_registry
+
+logging.basicConfig(level=logging.DEBUG)
+log = logging.getLogger(__name__)
+
+router = APIRouter()
+
+
+@router.get("/")
+async def get_tasks():
+    """
+    Returns all the available task and its schemas.
+
+    Returns
+    -------
+    List[JSON]
+        List of task JSONs
+    """
+
+    register_tasks = task_registry.registry
+    if not register_tasks:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="No tasks found"
+        )
+    all_tasks = defaultdict(dict)
+    for task_name, task_cls in register_tasks.items():
+        all_tasks[task_name] = task_cls.get_schema()
+    return all_tasks
```

Comment:
It is not necessary in this case to use defaultdict since you know the name of all the keys beforehand (and you are not using any object factory (tasks in this case) by default).

I propose this change using dict comprehensions:

```suggestion
    all_tasks = {
        task_name: task_cls.get_schema()
        for task_name, task_cls in register_tasks.items()
    }
    return all_tasks
```